src/Common/src/Common/ApplicationInstanceInfo.cs

```diff
@@ -44,7 +46,16 @@ public class ApplicationInstanceInfo : AbstractOptions, IApplicationInstanceInfo
 
         public string PlatformNameKey => BuildConfigString(PlatformRoot, ApplicationRoot + ":name");
 
-        protected static string PlatformRoot => string.Empty;
+        protected virtual string PlatformRoot => string.Empty;
+
+        protected void SecondChanceSetIdProperties(IConfiguration config = null)
```

Comment:
May be a comment on the constructors invoking this - rather than name it "second chance".